src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+
+            SetProperties(originalObj, deltaObject);
+
+            if (deltaObject.InstanceAnnotationsPropertyInfo != null) {
+
+                object instAnnValue;
+                changedObj.TryGetPropertyValue(deltaObject.InstanceAnnotationsPropertyInfo.Name, out instAnnValue);
+                IODataInstanceAnnotationContainer instanceAnnotations = instAnnValue as IODataInstanceAnnotationContainer;
+
+                if (instanceAnnotations != null)
+                {
+                    deltaObject.TrySetPropertyValue(deltaObject.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+                }
+            }
+
+            return deltaObject;
+        }
+
+        private void SetProperties(TStructuralType originalObj, Delta<TStructuralType> edmDeltaEntityObject)
+        {
+            foreach (string property in edmDeltaEntityObject.GetChangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+
+            foreach (string property in edmDeltaEntityObject.GetUnchangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+        }
+
+        private DeltaDeletedEntityObject<TStructuralType> CreateDeletedEntityForFailedOperation(Delta<TStructuralType> changedObj)
+        {
+            Type type = typeof(DeltaDeletedEntityObject<>).MakeGenericType(changedObj.ExpectedClrType);
+
+            DeltaDeletedEntityObject<TStructuralType> deletedObject = Activator.CreateInstance(type, true, changedObj.InstanceAnnotationsPropertyInfo) as DeltaDeletedEntityObject<TStructuralType>;
+
+            foreach (string property in changedObj.GetChangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            foreach (string property in changedObj.GetUnchangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            object annValue;
+            changedObj.TryGetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, out annValue);
+
+            IODataInstanceAnnotationContainer instanceAnnotations = annValue as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotations != null)
+            {
+                deletedObject.TrySetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+            }
+
+            deletedObject.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
+
+            TryGetContentId(changedObj, _keys, deletedObject);
+            
+            return deletedObject;
+        }
+
+        private static void TryGetContentId(Delta<TStructuralType> changedObj, IList<string> keys, DeltaDeletedEntityObject<TStructuralType> edmDeletedObject)
```

Comment:
>TryGetContentId [](start = 28, length = 15)

Can you add a comment to describe the logic and reasoning here? It looks like the logic is: if any key values are null, set the id of the deleted object to the contentId (or null if there is no contentId).  is this so ODL doesn't throw an exception when serializing and both key properties and id are null?